packages/coreutils/test/url.spec.ts

```diff
@@ -36,6 +36,13 @@ describe('@jupyterlab/coreutils', () => {
     describe('.join()', () => {
       it('should join a sequence of url components', () => {
         expect(URLExt.join('/foo/', 'bar/')).toBe('/foo/bar/');
+        expect(URLExt.join('//example.com', 'bar/')).toBe('//example.com/bar/');
+        expect(URLExt.join('//example.com', 'foo:bar/')).toBe(
+          '//example.com/foo:bar/'
+        );
+        expect(URLExt.join('http://www.example.com/', '/bar')).toBe(
+          'http://www.example.com/bar'
```

Comment:
Can you add these testcases as well?

```js
expect(URLExt.join('//example.com', 'foo:bar:', 'baz')).toBe('//example.com/foo:bar:/baz');
expect(URLExt.join('http://example.com', 'foo', '..', '..', 'bar')).toBe('http://example.com/bar/');
```

Reply:
I added those cases and found that the first one failed, so I also updated the code to address it. Specifically the replace shouldn't be called when there is no protocol

Reply:
Thanks Alex! I also looked at it a bit tonight, and realized we could be using the url-parse package like we are doing elsewhere in the file. If it's all right with you, can I submit a commit on top of this?

Reply:
Go ahead, my priority is just getting a fix in for our client